RELEASING.md

```diff
@@ -0,0 +1,39 @@
+- [Overview](#overview)
+- [Branching](#branching)
+  - [Release Branching](#release-branching)
+  - [Feature Branches](#feature-branches)
+- [Releasing](#releasing)
+- [Snapshots](#snapshot-builds)
+
+## Overview
+
+This document explains the release strategy for artifacts in this organization.
+
+## Branching
+
+### Release Branching
+
+Given the current major release of 0.1, projects in this organization maintain the following active branches.
+
+* **main**: The next _major_ release. This is the branch where all merges take place and code moves fast.
+* **1.x**: The next _minor_ release. Once a change is merged into `main`, decide whether to backport it to `1.x`.
+* **0.1**: The _current_ release. In between minor releases, only hotfixes (e.g. security) are backported to `0.1`.
```

Comment:
Not sure how this works. We had some pre-releases before https://github.com/opensearch-project/spring-data-opensearch/tree/0.1.0 where were main based. Since 1.0 is not released, pointing main to `2.0` does not make sense

Reply:
We have `1.x` on `1.0` and `main` on `2.0`. Since this is an experimental release we are releasing it under `0.1`.

Reply:
Until 1.0 is released, I believe that's the next major? As a best practice, main usually points to next major version which is `1.0` in this case since current release is `0.1`. Seems confusing a little.
Anyway it's just a suggestion and entirely upto the maintainers of this repository.

Reply:
Pointing `1.x` to `0.1.0` and let's release that.